Sage/carbon
# refactor(locales): convert files to typescript

### Proposed behaviour


Converts `date-fns-locale`, `locale`, `en-gb` and `pl-pl` to TypeScript

### Current behaviour


The `date-fns-locale`, `locale`, `en-gb` and `pl-pl` files are not TypeScript

### Checklist




#### QA


### Additional context



### Testing instructions


Check storybook translations still work for components that use the locale object (eg Date, Confirm etc)

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
